app/environments/brassbirmingham/brassbirmingham/envs/classes/board.py

```diff
@@ -423,6 +480,22 @@ def getAvailableCoalBuildingsTradePosts(
                     l.append(tradePost)
         return l
 
+    """
+    getAvailableIronBuildingsTradePosts
+    :return: buildings/tradeposts with iron"""
+
+    def getAvailableIronBuildingsTradePosts(
+        self
+    ) -> List[IndustryBuilding | TradePost]:
+        ironBuildings = self.getIronBuildings()
+        l = []
+        for ironBuilding in ironBuildings:
+            l.append(ironBuilding)
+        for tradePost in self.tradePosts:
+            if self.ironMarketRemaining > 0:
```

Comment:
Similar to comment above. `ironMarketRemaining` doesn't need be greater than 0. It's an unlimited suppy